src/server/project.ts

```diff
@@ -1901,16 +1901,20 @@ namespace ts.server {
                     compilerOptions,
                     moduleResolutionHost));
 
+                const program = hostProject.getCurrentProgram()!;
+                const symlinkCache = hostProject.getSymlinkCache();
                 for (const resolution of resolutions) {
                     if (!resolution.resolvedTypeReferenceDirective?.resolvedFileName) continue;
-                    const { resolvedFileName } = resolution.resolvedTypeReferenceDirective;
-                    const fileName = moduleResolutionHost.realpath?.(resolvedFileName) || resolvedFileName;
-                    if (!hostProject.getCurrentProgram()!.getSourceFile(fileName) && !hostProject.getCurrentProgram()!.getSourceFile(resolvedFileName)) {
-                        rootNames = append(rootNames, fileName);
+                    const { resolvedFileName, originalFileName } = resolution.resolvedTypeReferenceDirective;
+                    if (!program.getSourceFile(resolvedFileName) && (!originalFileName || !program.getSourceFile(originalFileName))) {
+                        rootNames = append(rootNames, resolvedFileName);
                         // Avoid creating a large project that would significantly slow down time to editor interactivity
                         if (dependencySelection === PackageJsonAutoImportPreference.Auto && rootNames.length > this.maxDependencies) {
                             return ts.emptyArray;
                         }
+                        if (originalFileName) {

```

Comment:
Does `discoverProbableSymlinks` need to handle symlinks from type reference resolution now that it happens (we do for module resolution)